File: travelAgency/Persistence/src/main/java/persistence/repository/TicketDBRepository.java
package persistence.repository;

import app.model.Ticket;
import persistence.TicketRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class TicketDBRepository implements TicketRepository {
    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public TicketDBRepository(Properties props) {
        this.dbUtils = new JdbcUtils(props);
    }
    @Override
    public Ticket findOne(Integer integer) {
        return null;
    }

    @Override
    public Iterable<Ticket> findAll() {
        return null;
    }

    @Override
    public Ticket save(Ticket entity) {
        logger.traceEntry( "saving task {}",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into tickets (flight_id, client_name, client_address, tourists_names, no_seats) values (?,?,?,?,?)")) {
            preStmt.setInt(1,entity.getId());
            preStmt.setString(2,entity.getClient_name());
            preStmt.setString(3,entity.getClientAddress());
            preStmt.setString(4,entity.getTourists_names());
            preStmt.setInt(5,entity.getNo_seats());
            int result=preStmt.executeUpdate();



            logger.trace("Saved () instances", result);

        }catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
            logger.traceExit();
            ex.printStackTrace();
        }
        return entity;
    }

    @Override
    public void delete(Integer integer) {
    }

    @Override
    public void update(Ticket entity) {

    }
}

File: Anul_2/Semestrul_2/Medii de proiectare si programare/mpp-proiect-csharp-andrea-muresan/TravelAgencyProject/Persistence/TicketDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using Model;
using Persistence;


namespace travelAgency.repository
{

    public class TicketDBRepository : ITicketRepository
    {
        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
        IDictionary<String, string> props;
        
        public TicketDBRepository(IDictionary<String, string> props)
        {
            log.Info("Creating TicketDBRepository");
            this.props = props;
        }

        public IEnumerable<Ticket> FindAll()
        {
            throw new NotImplementedException();
        }

        public Ticket FindOne(int id)
        {
            throw new NotImplementedException();
        }

        public void Save(Ticket entity)
        {
            log.InfoFormat("Saving ticket values: {0}, {1}, {2}", entity.Flight.Id, entity.ClientName, entity.NoSeats);
            IDbConnection con = DBUtils.getConnection(props);
            IList<Flight> ticketsR = new List<Flight>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText =
                    "INSERT INTO tickets(flight_id, client_name, client_address, tourists_names, no_seats) " +
                    "VALUES (@flightId, @client, @address, @tourists, @noSeats)";
                var paramFlightId = comm.CreateParameter();
                paramFlightId.ParameterName = "@flightId";
                paramFlightId.Value = entity.Flight.Id;
                comm.Parameters.Add(paramFlightId);
                
                var paramClient = comm.CreateParameter();
                paramClient.ParameterName = "@client";
                paramClient.Value = entity.ClientName;
                comm.Parameters.Add(paramClient);
                
                var paramAddress = comm.CreateParameter();
                paramAddress.ParameterName = "@address";
                paramAddress.Value = entity.ClientAddress;
                comm.Parameters.Add(paramAddress);
                
                var paramTourists = comm.CreateParameter();
                paramTourists.ParameterName = "@tourists";
                paramTourists.Value = entity.ClientsNames;
                comm.Parameters.Add(paramTourists);
                
                var paramNoSeats = comm.CreateParameter();
                paramNoSeats.ParameterName = "@noSeats";
                paramNoSeats.Value = entity.NoSeats;
                comm.Parameters.Add(paramNoSeats);

                int noVal = comm.ExecuteNonQuery();
                log.InfoFormat("{0} flights saved", noVal);
            }
        }

        public void Delete(int id)
        {
            throw new NotImplementedException();
        }

        public void Update(Ticket entity)
        {
            throw new NotImplementedException();
        }
    }
}